h2o-py/h2o/backend/server.py

```diff
@@ -264,7 +267,7 @@ def _jar_paths():
         yield os.path.join(prefix2, "h2o_jar", "h2o.jar")
 
     def _launch_server(self, port, baseport, mmax, mmin, ea, nthreads, jvm_custom_args, bind_to_localhost, 
-                       log_dir=None, log_level=None, max_log_file_size=None):
+                       log_dir=None, log_level=None, max_log_file_size=None, off_heap_memory_ratio=2.0/3):
```

Comment:
why do we replicate those default values all over the place client-side: 2x in py client, 1x in R, what about Flow?... can't we move this to backend?
Also is this 2/3 new default commonly agreed?

Reply:
The 2/3 is mentioned in the [docs](https://docs.h2o.ai/h2o/latest-stable/h2o-docs/data-science/xgboost.html#note).

The rationale is to show the warning to the user only if they can do something about it. And with the more advanced scenarios of running H2O, there is much more complexity, e.g., multi-node, gpu etc and the 2/3 rule might not apply so I went for the safer option.

Reply:
> The rationale is to show the warning to the user only if they can do something about it.

like messaging the admin that H2O fails when training XGB and he probably knows why? ;)

> And with the more advanced scenarios of running H2O, there is much more complexity, e.g., multi-node, gpu etc and the 2/3 rule might not apply so I went for the safer option.

as you're pointing out yourself, some of this complexity (any GPU?) also applies when starting from client. And we're not raising exception anywhere, so none of those default is safer than another.

More simply, seems to me that the idea is that we don't want to scare enterprise users with new warnings, but on the other side we want to raise the awareness of this memory concern for the single users.
But in that case, user should know what to do when he sees this warning:
- either, the warning should explain what to change, but as it's on the backend, it doesn't know the name of client's parameters and where they are set.
- or documentation should explain how to set this parameter for each client.
- and ideally, if it's just aimed at the basic client user, then the default should only be set in the main "public" method (`init` in py), the other methods are just forwarding methods receiving all args from init so we don't need to replicate the defaults there or change the "expert" behaviour there.